### PR TITLE
ENH add Omega_nu_mass to Omega_m

### LINKED ISSUE #647: Omega_m does not contain neutrinos
It just came to my attention that our Omega_m does not contain neutrinos.
I think this goes against standard convention in which Omega_m does contain massive neutrinos assuming they are fully non-relativistic and which is followed by at least Planck and BOSS papers (would be good to check, but I am 99% sure this is the case, because I remember fighting this cause on the "no neutrinos in Omega_m" side (let's call it yellow) and was convinced otherwise for Auburg et al BOSS paper). 
This would be an annoying break with the past, but on the other hand following different convention than Planck is also very dangerous. 
This should be at minimum discussed and clearly documented.
@damonge

### PR DESCRIPTION
**Pull request by beckermr**
We should include all mass-like species in Omega_m. 

Closes #647

### CODE REVIEW

File: pyccl/tests/test_parameters.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ def test_parameters_nu_list():
 
     assert np.allclose(cosmo['Omega_c'], 0.25)
     assert np.allclose(cosmo['Omega_b'], 0.05)
-    assert np.allclose(cosmo['Omega_m'], 0.30)
+    assert np.allclose(cosmo['Omega_m'] - cosmo['Omega_nu_mass'], 0.30)
```
Comment (elisachisari) [inline pyccl/tests/test_parameters.py]: When the user defines a value for Omega_m in the cosmology object, are they defining an Omega_m that includes massive neutrinos or not? I would assume that with the new approach the answer would be yes.

File: src/ccl_halofit.c [hunk 2]
```diff
@@ -24,7 +24,7 @@ void compute_chi(double a, ccl_cosmology *cosmo, double * chi, int * stat);
 
 static double zdrag_eh(ccl_parameters *params) {
   // eqn 4 of Eisenstein & Hu 1998
-  double OMh2 = params->Omega_m * params->h * params->h;
+  double OMh2 = (params->Omega_c + params->Omega_b) * params->h * params->h;
```
Comment (elisachisari) [inline src/ccl_halofit.c]: zdrag only enters in the EH calculation?

File: pyccl/tests/test_parameters.py (hunk 1 above)
Comment (beckermr) [reply pyccl/tests/test_parameters.py]: They don't ever do this. Omega_m value is computed.

File: src/ccl_halofit.c (hunk 2 above)
Comment (beckermr) [reply src/ccl_halofit.c]: yep this function is local to the halofit code.

File: src/ccl_background.c [hunk 3]
```diff
@@ -37,7 +37,7 @@ static double h_over_h0(double a, ccl_cosmology * cosmo, int *status)
              Om_mass_nu
   */
   return sqrt(
```
[the remark this answers is not in the document]
Comment (beckermr) [reply src/ccl_background.c]: We are not doing that. Omega_nu_mass is compute correctly below for each redshift and added correctly.

[the remark this answers is not in the document]
Comment (beckermr) [reply src/ccl_background.c]: see this `Om_mass_nu * a*a*a) / (a*a*a)`

**Comment by beckermr** [comment]
This PR does NOT remove the dependence on Omega_nu(z). However, by definition in CCL, Omega_m is the value at z = 0, which is what I set it to in the parameters. 

I did find one bug that I missed. Thanks for the review!

File: src/ccl_background.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (c-d-leonard) [reply src/ccl_background.c]: oops yeah you're right, sorry

File: src/ccl_core.c [hunk 4]
```diff
@@ -337,8 +337,8 @@ void ccl_parameters_fill_initial(ccl_parameters * params, int *status)
     params->Omega_nu_mass = 0.;
   }
 
-  params->Omega_m = params->Omega_b + params-> Omega_c;
-  params->Omega_l = 1.0 - params->Omega_m - params->Omega_g - params->Omega_nu_rel -params->Omega_nu_mass- params->Omega_k;
+  params->Omega_m = params->Omega_b + params-> Omega_c + params->Omega_nu_mass;
```
[the remark this answers is not in the document]
Comment (c-d-leonard) [reply src/ccl_core.c]: Got it, sorry about the confusion

**Comment by c-d-leonard** [comment]
Thanks for clarifying! Looks good to me.

**Comment by beckermr** [comment]
PR got too old and now a build is missing camb in the env

**Pull request merged**